README.md

```diff
@@ -43,93 +58,124 @@ interface reactDesc {
 
-__reactStamp.js__
+Let's go step by step through an example. It uses a pattern I call **Behavior Driven Composition**.
+
+__container.js__
 ```js
-import React from 'react';
-import reactStamp from 'react-stamp';
+container(React: object, components: object): componentDescriptor
+```
+```js
+export default (React, { Button, Text }) => ({
+  init() {
+    this.behaviors = [];
+    this.state = { showBehaviors: false };
+  },
+
+  render() {
+    const { showBehaviors } = this.state;
+    const behaviors = showBehaviors && this.behaviors.toString();
 
-export default reactStamp(React);
+    return (
+      <div>
+        <Button
+          value='Show Behaviors'
+          onClick={() => this._onClick()}
+        />
+        <Text
+          style={this.style}
+          behaviors={behaviors}
+        />
+      </div>
+    );
+  }
+});
 ```
 
-__component.js__
+BDC uses the concept of a container. The container is a HOC factory that defines the top level structure of the app. This particular container expects a `Button` component and a `Text` component. We only worry about designing the component signatures in the container, component behavior will be inherited later.
 
+__button.js__
 ```js
-import reactStamp from './reactStamp';
+export default React => (
+  ({ value, onClick }) => (
+    <input
+      type='button'
+      onClick={onClick}
+      value={value}
+    />
+  )
+);
+```
 
-export default reactStamp.compose({
-  state: {
-    obj1: false,
-    obj2: false,
-  },
+__text.js__
+```js
+export default React => (
+  ({ behaviors, style }) => (
+    <div style={style}>{behaviors}</div>
+  )
+);
+```
 
-  componentWillMount() {
-    this.setState({ obj1: true });
-  },
+BDC loves pure components. Pure components are simply [stateless functions](https://facebook.github.io/react/blog/2015/10/07/react-v0.14.html#stateless-functional-components). Notice that we are exporting factories that inject React. Read why [here](https://github.com/ericelliott/react-pure-component-starter#pure-component-factories). The function signatures are designed based on the component signatures defined in the container.
 
-  _onClick() {
-    return this.state;
+__buttonBehavior.js__
+```js
+export default {
+  init() {
+    this.behaviors.push('button');
   },
 
-  render() {
-    return (
-      <input
-        type='button'
-        onClick={() => this._onClick()}
-        value='Click Me'
-       />
-    );
+  _onClick () {
+    this.setState({
+      showBehaviors: !this.state.showBehaviors
+    });
   },
-});
+};
 ```
 
-__mixin.js__
-
+__textBehavior.js__
 ```js
 export default {
-  componentWillMount() {
-    this.setState({ obj2: true });;
+  init() {
+    this.behaviors.push('text');
+    this.style = { color: 'orange' }
   },
 };
 ```
 
-__app.js__
+The behavior mixins basically fill in the blanks. We see that `onClick`, `style`, and `behavior` have not been defined in the container. They get defined in behavior objects based on their behavior type.
```

Comment:
IMO any behaviour should be self-sufficient. But this code this will throw if you forget to initialize `this.behavior`. Other behaviours/components should be self-sufficient too.

If a behaviour/component depends on something it should "require" (or load somehow) the missing bit. Also IMO.